iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -47,8 +76,15 @@ void PoshRuntime::setRuntimeFactory(const factory_t& factory) noexcept
 
 PoshRuntime& PoshRuntime::defaultRuntimeFactory(cxx::optional<const RuntimeName_t*> name) noexcept
 {
-    static PoshRuntimeImpl instance(name);
-    return instance;
+    // Manual construction and destruction of the PoshRuntimeImpl, inspired by
+    // the nifty counter idiom.
+    static typename std::aligned_storage<sizeof(PoshRuntimeImpl), alignof(PoshRuntimeImpl)>::type buf;
+    static cxx::ScopeGuard ltp = [](auto name) {
```

Comment:
Can you think of a more descriptive name than `ltp`? Maybe `mainLifetimeParticipant` ... in better :)

Oh, it also seems I gave you a bad advice with capturing. Sorry for that. Didn't grasp that it's not a member.

Reply:
Ok, I named it `staticLifetimeParticipant` and added a comment.